
# HG changeset patch
# User Richard Dunbar <mojo@slitaz.org>
# Date 1509739839 14400
# Node ID ecb60a87e3e840843ad7698cee37380f374819ec
# Parent  40c8ad3478fdb0436487ff71966099a7546c7492
modules/install: PKG_SUM restores update new build detection

diff -r 40c8ad3478fd -r ecb60a87e3e8 modules/install
--- a/modules/install	Mon Oct 09 17:16:43 2017 +0100
+++ b/modules/install	Fri Nov 03 16:10:39 2017 -0400
@@ -620,7 +620,8 @@
 
 	# Remove newlines from some receipts
 	DEPENDS=$(echo $DEPENDS)
-
+	PKG_SUM="$(fgrep " $PACKAGE-$VERSION$EXTRAVERSION.tazpkg" "$PKGS_DB/installed.$SUM" | cut -d' ' -f1)"
+	
 	# Calculate "release checksum": md5sum of file containing md5sums of:
 	# a) all files, b) receipt, and c) description.txt.
 	rsumf=$(mktemp)
@@ -637,7 +638,7 @@
 	sed -i "/^$PACKAGE	/d" "$ii"
 
 	cat >> "$ii" <<EOT
-$PACKAGE	$VERSION$EXTRAVERSION	$CATEGORY	$SHORT_DESC	$WEB_SITE	$TAGS	$SIZES	$DEPENDS	$RSUM
+$PACKAGE	$VERSION$EXTRAVERSION	$CATEGORY	$SHORT_DESC	$WEB_SITE	$TAGS	$SIZES	$DEPENDS	$PKG_SUM
 EOT
 
 	TEMP_FILE="$(mktemp)"

